libs/ebpfnetsh/programs.cpp

```diff
@@ -78,7 +79,13 @@ struct _program_unloader
 
 unsigned long
 handle_ebpf_add_program(
-    LPCWSTR machine, LPWSTR* argv, DWORD current_index, DWORD argc, DWORD flags, LPCVOID data, BOOL* done)
+    _Null_terminated_ const wchar_t* machine,
+    _Null_terminated_ wchar_t** argv,
+    unsigned long current_index,
+    unsigned long argc,
+    unsigned long flags,
+    const void* data,
+    BOOL* done)
```

Comment:
this would be ```bool*```.  Other instances in this PR would need to be fixed as well.

Reply:
I don't think it's bool which is a C++ only type, it has to match the C header type where I think BOOL != bool. That said, yes it would be good to replace BOOL by whatever the underlying C type is.

Reply:
Agree that C does not have the ```bool``` type, but we do use ```bool``` in our existing code base (```ebpf_core_object_map_t``` struct definition, for example), so thought maintaining that consistency would be preferrable.

FYI: (```bool``` is #defined in ```ebpf-for-windows\libs\platform\kernel\framework.h```)